src/plugins/cjs-interop.ts

```diff
@@ -10,17 +15,81 @@ export const cjsInterop = (): Plugin => {
         this.format !== 'cjs' ||
         info.type !== 'chunk' ||
         !/\.(js|cjs)$/.test(info.path) ||
-        !info.entryPoint ||
-        info.exports?.length !== 1 ||
-        info.exports[0] !== 'default'
+        !info.entryPoint
       ) {
         return
       }
 
+      if (this.splitting) {
+        // there is exports metadata when cjs+splitting is set
+        if (info.exports?.length !== 1 || info.exports[0] !== 'default') return
+      } else {
+        const swc: typeof import('@swc/core') = localRequire('@swc/core')
+        const { Visitor }: typeof import('@swc/core/Visitor') =
+          localRequire('@swc/core/Visitor')
+        if (!swc || !Visitor) {
+          throw new PrettyError(
+            `@swc/core is required for cjsInterop when splitting is not enabled. Please install it with \`npm install @swc/core -D\``
+          )
+        }
+
+        let entrySource: string | undefined
+        try {
+          entrySource = await fs.readFile(info.entryPoint!, {
+            encoding: 'utf8',
+          })
+        } catch {}
+        if (!entrySource) return
+
+        const ast = await swc.parse(entrySource, {
+          syntax: 'typescript',
+          decorators: true,
+          tsx: true,
+        })
+        const visitor = createExportVisitor(Visitor)
+        visitor.visitProgram(ast)
+
+        if (
+          !visitor.hasExportDefaultExpression ||
+          visitor.hasNonDefaultExportDeclaration
+        )
+          return
+      }
+
       return {
-        code: code + '\nmodule.exports = exports.default;\n',
+        code: code + '\nmodule.exports=module.exports.default;\n',
```

Comment:
Why to remove spaces?

Reply:
1. The process of minification is complete before we reach here. This keeps the output compact
2. The main point of this modification is
```diff
- module.exports = exports.default
+ module.exports = module.exports.default
```

the test case fails without it:
```ts
await expect(
  runCjsInteropTest('simple', {
    'input.ts': `export default { hello: 'world' }`,
  })
).resolves.toEqual({ hello: 'world' })
```